### PR TITLE
Enable Nuget caching locally

### PR DESCRIPTION
**Pull request by emabrey**
<details>
<summary>Checklist</summary>

- [x] I have signed off my commits using `-s` or `Signed-off-by`\* (See: [Contributing § DCO](https://github.com/tenacityteam/tenacity/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code\*
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving\*
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"\*

\* indicates required

</details>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -133,6 +133,36 @@ if( VCPKG )
...
+      COMMAND vcpkg fetch nuget
+      WORKING_DIRECTORY ${VCPKG_ROOT}
+      OUTPUT_VARIABLE NUGET_EXECUTABLE
+      OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+
+    message( STATUS "Automatically setting Nuget api key to allow nuget cache usage locally" )
+
+    file( READ ${CMAKE_SOURCE_DIR}/nuget.config.in NUGET_CONFIG_CONTENT )
+    file( WRITE ${CMAKE_SOURCE_DIR}/nuget.config ${NUGET_CONFIG_CONTENT} )
```
Comment (Be-ing) [inline CMakeLists.txt:159]: This is just copying the file? Why not use `file(COPY)`?

Comment (emabrey) [reply CMakeLists.txt:159]: It has to have a different name.

**Comment by Be-ing** [comment]
Is this working for you locally?

**Comment by n0toose** [comment]
Works locally on Arch Linux.

**Comment by n0toose** [comment]
Okay, this is seriously ***amazing*** stuff right here.

**Review by n0toose — DISMISSED** [review]
YES.

**Comment by Be-ing** [comment]
@emabrey if you confirm this is working locally on Windows, go ahead and merge.

**Pull request merged**